src/main/java/org/dasein/cloud/network/VPNLaunchOptions.java

```diff
@@ -0,0 +1,73 @@
+package org.dasein.cloud.network;
+
+import java.util.Arrays;
+
+import javax.annotation.Nonnull;
+
+public class VPNLaunchOptions {
+
+    private String providerVlanId = null;
+    private String providerDataCenterId = null;
+    private String[] labels = new String[0];
+    private String name = null;
+    private String description = null;
+    private VPNProtocol protocol = null;
+
+    private VPNLaunchOptions() { }
+
+    private VPNLaunchOptions(@Nonnull String name, @Nonnull String description, @Nonnull VPNProtocol protocol) {
+        this.name = name;
+        this.description = description;
+        this.protocol = protocol;
+    }
+
+    static public @Nonnull VPNLaunchOptions getInstance(@Nonnull String name, @Nonnull String description, @Nonnull VPNProtocol protocol) {
+        return new VPNLaunchOptions(name, description, protocol);
+    }
+
+    /**
+     * Specifies any labels to be added to the VPN
+     * @param labels one or more labels to be added to new VPN
+     * @return this
+     */
+    public @Nonnull VPNLaunchOptions withLabels(String... labels) {
+        if (labels != null) {
+            this.labels  = Arrays.copyOf(labels, labels.length);
+        }
+        return this;
+    }
+
+    public @Nonnull VPNLaunchOptions withProviderVlanId(@Nonnull String providerVlanId) {
+        this. providerVlanId = providerVlanId;
```

Comment:
Same here `VPN` but `Vlan`. We have this all over the place, and I plan to make it all uniform, i.e. `Vpn` and `Vlan`.